lib/child_process.js

```diff
@@ -329,6 +329,49 @@ function normalizeSpawnArguments(file /*, args, options*/) {
   else if (options === null || typeof options !== 'object')
     throw new TypeError('"options" argument must be an object');
 
+  // Validate the cwd, if present.
+  if (options.cwd !== undefined &&
+      options.cwd !== null &&
+      !typeof options.cwd === 'string')
```

Comment:
Do you mean another function like `isOptionalString()` (where the optional part covers `null` and `undefined`)?